ols/src/llms/providers/rhoai_vllm.py

```diff
@@ -0,0 +1,54 @@
+"""Red Hat OpenShift VLLM provider implementation."""
+
+import logging
+from typing import Any, Optional
+
+from langchain.llms.base import LLM
+from langchain_openai import ChatOpenAI
+
+from ols import constants
+from ols.src.llms.providers.provider import LLMProvider
+from ols.src.llms.providers.registry import register_llm_provider_as
+
+logger = logging.getLogger(__name__)
+
+
+@register_llm_provider_as(constants.PROVIDER_RHOAI_VLLM)
+class RHOAIVLLM(LLMProvider):
+    """RHOAI VLLM provider."""
+
+    # note: there's no default URL for RHOAI VLLM
+    url: str = "https://api.openai.com/v1"
+    credentials: Optional[str] = None
+
+    @property
+    def default_params(self) -> dict[str, Any]:
+        """Default LLM params."""
+        self.url = str(self.provider_config.url or self.url)
+        self.credentials = self.provider_config.credentials
+        # provider-specific configuration has precendence over regular configuration
+        if self.provider_config.rhoai_vllm_config is not None:
+            rhoai_vllm_config = self.provider_config.rhoai_vllm_config
+            self.url = str(rhoai_vllm_config.url)
+            if rhoai_vllm_config.api_key is not None:
+                self.credentials = rhoai_vllm_config.api_key
+
+        return {
+            "base_url": self.url,
+            "openai_api_key": self.credentials,
+            "model": self.model,
+            "model_kwargs": {
+                "top_p": 0.95,
+                "frequency_penalty": 1.03,
+            },
+            "organization": None,
+            "cache": None,
+            "streaming": True,
+            "temperature": 0.01,
+            "max_tokens": 512,
```

Comment:
is 512 tokens too short for generating deployment YAML files?

Reply:
in this case 512 tokens is approx 280-300 characters (only). So might not be enough